dot/import.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2019 ChainSafe Systems (ON) Corp.
+// This file is part of gossamer.
+//
+// The gossamer library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The gossamer library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
+
+package dot
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"math/big"
+	"path/filepath"
+
+	"github.com/ChainSafe/gossamer/dot/state"
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/trie"
+
+	log "github.com/ChainSafe/log15"
+)
+
+// ImportState imports the state in the given files to the database with the given path.
+func ImportState(basepath, stateFP, headerFP string, firstSlot uint64) error {
+	tr, err := newTrieFromPairs(stateFP)
+	if err != nil {
+		return err
+	}
+
+	header, err := newHeaderFromFile(headerFP)
+	if err != nil {
+		return err
+	}
+
+	log.Info("ImportState", "header", header)
+
+	srv := state.NewService(basepath, log.LvlInfo)
+	return srv.Import(header, tr, firstSlot)
+}
+
+func newTrieFromPairs(filename string) (*trie.Trie, error) {
+	data, err := ioutil.ReadFile(filepath.Clean(filename))
+	if err != nil {
+		return nil, err
+	}
+
+	pairs := make([]interface{}, 0)
+	err = json.Unmarshal(data, &pairs)
+	if err != nil {
+		return nil, err
+	}
+
+	entries := make(map[string]string)
+	for _, pair := range pairs {
+		pairArr := pair.([]interface{})
+		entries[pairArr[0].(string)] = pairArr[1].(string)
```

Comment:
Can we assert as a safety check that the length of pairArr is always equal to two?